src/test/e2e/api/client/feature.optimal304.e2e.test.ts

```diff
@@ -1,220 +1,314 @@
         }
     });
 
-    test('returns 200 when content updates and hash does not match anymore', async () => {
-        await app.services.featureToggleService.createFeatureToggle(
-            'default',
-            {
-                name: `featureNew304${apendix}`,
-                description: 'the #1 feature',
-            },
-            TEST_AUDIT_USER,
-        );
+    test('creating a new feature does not modify etag', async () => {
+        await app.createFeature('new');
         await app.services.configurationRevisionService.updateMaxRevisionId();
 
-        const res = await app.request
+        await app.request
             .get('/api/client/features')
-            .set('if-none-match', 'ae443048:16')
+            .set('Authorization', devTokenSecret)
+            .set('if-none-match', `"76d8bb0e:13${enabled ? `:${name}` : ''}"`)
+            .expect(304);
+    });
+
+    test('a token with all envs should get the max id regardless of the environment', async () => {
+        const currentProdEtag = `"67e24428:15${enabled ? `:${name}` : ''}"`;
+        const { headers } = await app.request
+            .get('/api/client/features')
+            .set('if-none-match', currentProdEtag)
+            .set('Authorization', allEnvsTokenSecret)
             .expect(200);
 
-        if (etagVariant.feature_enabled) {
-            expect(res.headers.etag).toBe(`"76d8bb0e:16:${etagVariant.name}"`);
-            expect(res.body.meta.etag).toBe(
-                `"76d8bb0e:16:${etagVariant.name}"`,
-            );
-        } else {
-            expect(res.headers.etag).toBe('"76d8bb0e:16"');
-            expect(res.body.meta.etag).toBe('"76d8bb0e:16"');
-        }
+        // it's a different hash than prod, but gets the max id
+        expect(headers.etag).toEqual(
+            `"ae443048:15${enabled ? `:${name}` : ''}"`,
+        );
     });
-});
 
-// running after all inside describe block, causes some of the queries to fail to acquire a connection
-// this workaround is to run the afterAll outside the describe block
-afterAll(async () => {
-    await Promise.all(shutdownHooks.map((hook) => hook()));
+    test('production environment gets a different etag than development', async () => {
+        const { headers: prodHeaders } = await app.request
+            .get('/api/client/features?bla=1')
+            .set('Authorization', prodTokenSecret)
+            .expect(200);
+
+        expect(prodHeaders.etag).toEqual(
+            `"67e24428:15${enabled ? `:${name}` : ''}"`,
+        );
+
+        const { headers: devHeaders } = await app.request
+            .get('/api/client/features')
+            .set('Authorization', devTokenSecret)
+            .expect(200);
+
+        expect(devHeaders.etag).toEqual(
+            `"76d8bb0e:13${enabled ? `:${name}` : ''}"`,
+        );
+    });
+
+    test('modifying dev environment should only invalidate dev tokens', async () => {
```

Comment:
This test validates that modifying a flag in a single environment only invalidates that environment.